build.Dockerfile

```diff
@@ -0,0 +1,35 @@
+FROM rust:bullseye
+SHELL ["/bin/bash", "-c"]
+RUN set -ex; \
+    \
+    dpkg --add-architecture amd64; \
+    dpkg --add-architecture arm64; \
+    \
+    apt-get update; \
+    apt-get install -y --no-install-recommends \
+        binutils-{aarch64,x86-64}-linux-gnu \
+        build-essential \
+        crossbuild-essential-{amd64,arm64} \
+        debhelper-compat \
+        devscripts \
+        dh-cargo \
+        dh-make \
+        libssl-dev:{amd64,arm64} \
+        lintian \
+        # Necessary for automatic dependency resolution.
+        {libc6,libssl1.1}:{amd64,arm64} \
+    ; \
+    rm -rf /var/lib/apt/lists/*
+# Prevents unnecessary toolchain downloads.
+WORKDIR /mnt
+RUN --mount=source=rust-toolchain.toml,target=/mnt/rust-toolchain.toml \
+    \
+    rustup target add {aarch64,x86_64}-unknown-linux-gnu
+RUN set -ex; \
+    \
+    # Use image-provided Cargo, as Bullseye doesn't provide Rustup.
+    ln -fs /usr/local/cargo/bin/cargo /usr/bin/cargo; \
+    # Prevent using nightly-only features.
+    sed -i 's/"-Zavoid-dev-deps", //g' /usr/share/cargo/bin/cargo; \
```

Comment:
I'm failing to understand what it can prevent. AFAICT, debian's wrapper around cargo `/usr/share/cargo/bin/cargo` is not in use at all. How does it prevent using nightly-only features?

Reply:
`/usr/share/cargo/bin/cargo` is used by `dh-cargo` in `/usr/share/perl5/Debian/Debhelper/Buildsystem/cargo.pm`. This command removes all occurrences of `"-Zavoid-dev-deps", `, which according to https://doc.rust-lang.org/cargo/reference/unstable.html#avoid-dev-deps is still unstable, from `/usr/share/cargo/bin/cargo`.

You can also prove this yourself by removing this command. Cargo calls would fail due to `-Zavoid-dev-deps` being unrecognized.

Reply:
Ok, no problem then.

Was there any discussion about using/not using cargo deb?